blockchain/vm/internaltx_tracer.go

```diff
@@ -0,0 +1,498 @@
+}
+
+// result is invoked when all the opcodes have been iterated over and returns
+// the final result of the tracing.
+func (this *InternalTxTracer) result() (*InternalTxTraceResult, error) {
+	if _, exist := this.ctx["type"]; !exist {
+		this.ctx["type"] = OpCode(0)
+	}
+	if _, exist := this.ctx["from"]; !exist {
+		this.ctx["from"] = common.Address{}
+	}
+	if _, exist := this.ctx["to"]; !exist {
+		this.ctx["to"] = common.Address{}
+	}
+	if _, exist := this.ctx["value"]; !exist {
+		this.ctx["value"] = big.NewInt(0)
+	}
+	if _, exist := this.ctx["gas"]; !exist {
+		this.ctx["gas"] = uint64(0)
+	}
+	if _, exist := this.ctx["gasUsed"]; !exist {
+		this.ctx["gasUsed"] = uint64(0)
+	}
+	if _, exist := this.ctx["input"]; !exist {
+		this.ctx["input"] = ""
+	}
+	if _, exist := this.ctx["output"]; !exist {
+		this.ctx["output"] = ""
+	}
+	if _, exist := this.ctx["time"]; !exist {
+		this.ctx["time"] = time.Duration(0)
+	}
+	if this.callStackLength() == 0 {
+		this.callStack = []*InternalCall{{}}
+	}
+	result := &InternalTxTraceResult{
+		CallType: this.ctx["type"].(OpCode),
+		From:     this.ctx["from"].(common.Address),
+		To:       this.ctx["to"].(common.Address),
+		Value:    "0x" + this.ctx["value"].(*big.Int).Text(16),
+		Gas:      this.ctx["gas"].(uint64),
+		GasUsed:  this.ctx["gasUsed"].(uint64),
+		Input:    this.ctx["input"].(string),
+		Output:   this.ctx["output"].(string),
+		Time:     this.ctx["time"].(time.Duration),
+	}
+	if this.callStack[0].calls != nil {
+		result.Calls = this.callStack[0].calls
+	}
+	if this.callStack[0].Err != nil {
+		result.Error = this.callStack[0].Err
+	} else if ctxErr, _ := this.ctx["error"]; ctxErr != nil {
+		result.Error = ctxErr.(error)
+	}
+	if result.Error != nil {
+		result.Output = "" // delete result.output;
+	}
+	if this.ctx["error"] == errEvmExecutionReverted {
+		outputHex := this.ctx["output"].(string) // it is already a hex string
+		if outputHex[2:10] == "08c379a0" {
+			defaultOffset := 10
+
+			stringOffset, err := strconv.ParseInt(outputHex[defaultOffset:defaultOffset+32*2], 16, 64)
+			if err != nil {
+				logger.Error("failed to parse hex string to get stringOffset",
+					"err", err, "outputHex", outputHex)
+				return nil, err
+			}
+			stringLength, err := strconv.ParseInt(outputHex[defaultOffset+32*2:defaultOffset+32*2+32*2], 16, 64)
+			if err != nil {
+				logger.Error("failed to parse hex string to get stringLength",
+					"err", err, "outputHex", outputHex)
+				return nil, err
+			}
+			start := defaultOffset + 32*2 + int(stringOffset*2)
+			end := start + int(stringLength*2)
+			asciiInBytes, err := hex.DecodeString(outputHex[start:end])
+			if err != nil {
+				logger.Error("failed to parse hex string to get ASCII representation",
+					"err", err, "outputHex", outputHex)
+				return nil, err
+			}
+			this.revertString = string(asciiInBytes)
+		}
+		result.Reverted = RevertedInfo{Contract: this.revertedContract, Message: this.revertString}
+	}
+	return result, nil
```

Comment:
In `call_tracer.js`, `finalize` is called before returning in this line. But I cannot find such logic. Is this intended? :)

Reply:
Yes, it is intended as what it does is just to prettify the printed result.